wikibase.api.js

```diff
@@ -28,51 +71,39 @@ class WikibaseApi {
 
 		Object.assign( itemData, { labels }, data );
 
-		const bot = new MWBot( {
-			apiUrl: `${browser.config.baseUrl}/api.php`
-		} );
+		const bot = await this.getBot();
 
-		return bot.getEditToken()
-			.then( () => {
-				return bot.request( {
-					action: 'wbeditentity',
-					new: 'item',
-					data: JSON.stringify( itemData ),
-					token: bot.editToken
-				} );
-			} ).then( ( response ) => {
-				return response.entity.id;
-			} );
+		return bot.request( {
+			action: 'wbeditentity',
+			new: 'item',
+			data: JSON.stringify( itemData ),
+			token: bot.editToken
+		} ).then( ( response ) => {
+			return response.entity.id;
+		} );
 	}
 
-	createProperty( datatype, data ) {
+	async createProperty( datatype, data ) {
 		let propertyData = {};
 
 		propertyData = Object.assign( {}, { datatype }, data );
 
-		const bot = new MWBot( {
-			apiUrl: `${browser.config.baseUrl}/api.php`
-		} );
+		const bot = await this.getBot();
 
-		return bot.getEditToken()
-			.then( () => {
-				return new Promise( ( resolve, reject ) => {
-					bot.request( {
-						action: 'wbeditentity',
-						new: 'property',
-						data: JSON.stringify( propertyData ),
-						token: bot.editToken
-					} ).then( ( response ) => {
-						resolve( response.entity.id );
-					}, reject );
-				} );
-			} );
+		return new Promise( ( resolve, reject ) => { // FIXME: I don't think this Promise is needed
```

Comment:
I don’t think the Promise is needed either, we can probably just await the response and then `return response.entity.id`? (And I’d say the same for `createItem()` – once we start using async/await, it seems unnecessary to keep working with `.then()` “manually”.)

Reply:
Yes. Though I feel like I want to split up the change to how we use mwbot and general Promise=>async/await cleanup into two commits